src/github.com/couchbaselabs/sync_gateway/base/bucket.go

```diff
@@ -77,6 +85,23 @@ func (bucket couchbaseBucket) View(ddoc, name string, params map[string]interfac
 }
 
 func (bucket couchbaseBucket) StartTapFeed(args walrus.TapArguments) (walrus.TapFeed, error) {
+	// Use tap only when it's explicitly requested, or if DCP isn't supported
+	if bucket.spec.FeedType == "tap" {
+		LogTo("Feed", "Using TAP feed for bucket: %q (based on feed_type specified in config file", bucket.GetName())
+		return bucket.StartCouchbaseTapFeed(args)
+	} else {
+		feed, err := bucket.StartDCPFeed(args)
+		if err != nil {
+			Warn("Unable to start DCP feed - reverting to using TAP feed: %s", err)
+			return bucket.StartCouchbaseTapFeed(args)
+		} else {
+			LogTo("Feed", "Using DCP feed for bucket: %q", bucket.GetName())
+			return feed, nil
+		}
+	}
+}
+
+func (bucket couchbaseBucket) StartCouchbaseTapFeed(args walrus.TapArguments) (walrus.TapFeed, error) {
```

Comment:
The naming is a bit confusing, having both `StartTapFeed` and `StartCouchbaseTapFeed`.  (is `StartTapFeed` starting a non-couchbase tap feed?  what does that mean .. does that mean its a Walrus tap feed?)